src/hooks/useForm.js

```diff
@@ -0,0 +1,83 @@
+import { useState, useCallback } from 'react';
+
+const useForm = (
+  {
+    initialValues,
+    onSubmit,
+    validator = () => {},
+    validateOnChange = false,
+    validateOnBlur = false,
+  },
+  ...dependencies
+) => {
+  const [values, setValues] = useState(initialValues);
+  const [errors, setErrors] = useState({});
+  const [blurredFields, setBlurredFields] = useState({});
+
+  const handleSubmit = useCallback(() => {
+    const newErrors = validator(values) || {};
+    const valid = !Object.values(newErrors)
+      .filter(error => !!error)
+      .reduce((acc, error) => {
+        error.forEach(e => acc.push(e));
```

Comment:
why is this necessary? i'm not following it

Reply:
This flattens the errors for each field, since you could have more than one.